brion/plugin/morphologySWC.cpp

```diff
@@ -80,15 +76,24 @@ struct Sample
     bool valid;
     Vector4f point; // x, y, z and diameter
     SWCSectionType type;
-    int parent;
-    int nextID;
-    int siblingID;
-    int parentSection; // Only meaningful for the first sample of each
-                       // section
+    int parent = -1;        // sample index
+    int nextID = -1;        // sample index
+    int siblingID = -1;     // sample index
+    int sectionID = -1;     // Only necessary for the last sample of a section
+    int parentSection = -1; // Only meaningful for the first sample of a section
+};
+using Samples = std::vector<Sample>;
+auto _comparator = [](const Sample* a, const Sample* b) {
+    // We want undefined sections to go last but we need to respect the
+    // numerical values from the swc "specification".
+    // The conversion of -1 to size_t takes care of this
+    return (uint32_t(a->type - 1) > uint32_t(b->type - 1) ||
+            (a->type == b->type && a > b));
```

Comment:
Why do you use pointer comparison in case of equality ?

Reply:
Good question, that should be documented. The address reflects which sample was parsed first from the file, which is exactly the sorting condition that we have to apply to follow the spec.